app/reducers/fetchHistory.js

```diff
@@ -1,26 +1,59 @@
 // @flow
 
 import { FetchHistory, User } from 'app/actions/ActionTypes';
+import { omit } from 'lodash';
 
 type State = {};
 
 const initialState = {};
+export const fetchHistoryEntryKey = (meta: {
+  paginationKey: string,
+  cursor: string,
+  endpoint: string,
+}) =>
+  // Add random (but stable) string before endpoint so that we don't mix actions with paginationKey and those without
+  meta &&
+  (meta.paginationKey
+    ? 'paginationKeyAndCursor:' + meta.paginationKey + '&cursor=' + meta.cursor
+    : meta.endpoint);
```

Comment:
Thought about adding query to the key when using `meta.endpoint` as well, but this cursor is from the internal pagination (paginationNext), and with the old pagination stuff it is hard to calculate the query since it is used in a few different ways.

So yeah, this "works" for all kinds of pagination, but only works "perfect" for the new style.

Reply:
I feel like this is the "correct" way of doing it, since we shouldn't need to fix this for legacy stuff. It will just get more confusing. There's no point in making it work with actions not using `paginationKey`, and it should give some more incentive to migrate old stuff so this actually works with it as well :sweat_smile:.